pkg/server/webhook.go

```diff
@@ -134,20 +134,29 @@ func (whsvr *WebhookServer) requiredMutation(ignoredList []string, metadata *met
 		annotations = map[string]string{}
 	}
 
-	status := annotations[whsvr.Config.AnnotationNamespace+"/status"]
+	statusAnnotationKey := whsvr.Config.AnnotationNamespace + "/status"
+	requestAnnotationKey := whsvr.Config.AnnotationNamespace + "/request"
+
+	status, ok := annotations[statusAnnotationKey]
+	if ok && strings.ToLower(status) == "injected" {
+		glog.Infof("Pod %s/%s annotation %s=%s indicates injection already satisfied, skipping.", metadata.Namespace, metadata.Name, statusAnnotationKey, status)
+		return "", ErrSkipAlreadyInjected
+	}
 
 	// determine whether to perform mutation based on annotation for the target resource
-	requestedInjection := ""
-	if strings.ToLower(status) != "injected" {
-		requestedInjection = strings.ToLower(annotations[whsvr.Config.AnnotationNamespace+"/request"])
-		if !whsvr.Config.HasInjectionConfig(requestedInjection) {
-			glog.Errorf("Mutation policy for %v/%v: status:%q requested injection: %s was not in configuration, skipping", metadata.Namespace, metadata.Name, status, requestedInjection)
-			return ""
-		}
+	requestedInjection, ok := annotations[requestAnnotationKey]
+	injectionKey := strings.ToLower(requestedInjection)
+	if !ok {
+		glog.Infof("Pod %s/%s annotation %s is missing, skipping injection.", metadata.Namespace, metadata.Name, statusAnnotationKey)
+		return requestedInjection, ErrMissingRequestAnnotation
```

Comment:
Might want to `return "", ErrMissingRequestAnnotation` for clarity as well.

Reply:
Oops, good catch!